src/RosInterface.cpp

```diff
@@ -236,6 +237,35 @@ namespace phoxi_camera {
         return true;
     }
 
+    bool RosInterface::getScanningVolume(phoxi_camera::GetScanningVolume::Request& req, phoxi_camera::GetScanningVolume::Response& res) {
+        try {
+            pho::api::PhoXiScanningVolume scanVolume = scanner->ScanningVolume;
+            pho::api::PhoXiMesh phoxiMesh = scanVolume.Mesh;
+            pcl::PointCloud<pcl::PointXYZ> meshPointCloud{};
+            pcl::PolygonMesh meshPcl;
+
+            for (auto point : phoxiMesh.Vertices) {
+                meshPointCloud.push_back(pcl::PointXYZ(point.x, point.y, point.z));
+            }
+            pcl::toPCLPointCloud2(meshPointCloud, meshPcl.cloud);
+
+            uint numOfPolygons = phoxiMesh.Indices.size()/3;
+            meshPcl.polygons.resize(numOfPolygons);
+            for (int i = 0; i < (numOfPolygons); i++) {
+                for (int v = 0; v < 3; v++) {
+                    meshPcl.polygons[i].vertices.push_back(phoxiMesh.Indices[i * 3 + v]);
+                }
+            }
+            pcl_conversions::moveFromPCL(meshPcl, res.mesh);
+            res.success = true;
+            res.message = OKRESPONSE;
+        } catch (PhoXiInterfaceException& e) {
+            res.success = false;
+            res.message = e.what();
+        }
+        return true;
```

Comment:
I would appreciate if you could specify what kind of conversion is happening here. Why we can't transfer the pho::api::PhoXiMesh directly to pcl_msgs::PolygonMesh?

Reply:
Encapsulated and slightly changed for better readability.